internal/services/nginx/nginx_deployment_data_source_test.go

```diff
@@ -42,3 +43,30 @@ data "azurerm_nginx_deployment" "test" {
 }
 `, DeploymentResource{}.basic(data))
 }
+
+func (d NginxDeploymentDataSource) basicAutoscaling(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+data "azurerm_nginx_deployment" "test" {
+  name                = azurerm_nginx_deployment.test.name
+  resource_group_name = azurerm_nginx_deployment.test.resource_group_name
+}
+`, DeploymentResource{}.basicAutoscaling(data))
+}
+
+func TestAccNginxDeploymentDataSource_autoscaling(t *testing.T) {
```

Comment:
I'm not familiar with these testing helpers; how hard would it be to add a test showing that using both `capacity` and `auto_scale_profile` returns an error?